x-pack/plugin/runtime-fields/src/main/java/org/elasticsearch/xpack/runtimefields/query/AbstractBooleanScriptFieldQuery.java

```diff
@@ -41,35 +33,6 @@
      */
     protected abstract boolean matches(int trues, int falses);
 
-    @Override
-    public final Weight createWeight(IndexSearcher searcher, ScoreMode scoreMode, float boost) throws IOException {
-        return new ConstantScoreWeight(this, boost) {
-            @Override
-            public boolean isCacheable(LeafReaderContext ctx) {
-                return false; // scripts aren't really cacheable at this point
-            }
-
-            @Override
-            public Scorer scorer(LeafReaderContext ctx) throws IOException {
-                BooleanScriptFieldScript script = leafFactory.newInstance(ctx);
-                DocIdSetIterator approximation = DocIdSetIterator.all(ctx.reader().maxDoc());
-                TwoPhaseIterator twoPhase = new TwoPhaseIterator(approximation) {
-                    @Override
-                    public boolean matches() throws IOException {
-                        script.runForDoc(approximation().docID());
-                        return AbstractBooleanScriptFieldQuery.this.matches(script.trues(), script.falses());
-                    }
-
-                    @Override
-                    public float matchCost() {
-                        return MATCH_COST;
-                    }
-                };
-                return new ConstantScoreScorer(this, score(), scoreMode, twoPhase);
-            }
-        };
-    }
-
     @Override
     public final void visit(QueryVisitor visitor) {
         // No subclasses contain any Terms because those have to be strings.
```

Comment:
I was also wondering if the visit methods are consistent, i see that most of the impl share the same code, but there are subtleties as not all impls override it. I was about to move the standard impl to the base class but I refrained.

Reply:
The ones that are not for strings aren't going to emit any terms so they all get the standard impl. The String ones have much more interesting `visit` methods.

Reply:
We could end up with a `AbstractNonStringRuntimeFieldQuery` or something like that which had the base class. I didn't go for that because I didn't think it was worth the extra layer just for that one class. But if there are more classes it'd make sense.

Reply:
ok I will have another look later.